Kui-kui/aircall-frontend-hiring-test
# [MASTER] feat: implement call archiving + real-time support

# 🔍 Scope

- On the call detail page, I added a button to either archive or restore a call (depending on the current value of `call.is_archived`)
- I also set up Apollo subscriptions to enable real-time support on call updates

# 📷 Screenshots

## Archive a call 

<img width="1438" alt="Capture d’écran 2023-01-21 à 17 38 21" src="https://user-images.githubusercontent.com/14906916/213877146-dd64dbab-3b09-44dd-b2f4-1b80bad314be.png">

## Restore an archived call

<img width="1438" alt="Capture d’écran 2023-01-21 à 17 38 27" src="https://user-images.githubusercontent.com/14906916/213877156-a820c01a-4f7b-4d38-95fe-ace27d1b0007.png">

## Real-time support

https://user-images.githubusercontent.com/14906916/213877365-a822627d-0f98-4f1a-8c88-cfb7f2ab6d14.mp4

# 🧪 Test steps

1. Open the app in two different tabs 
2. Login on both tabs
3. Select any call in the list, but make sure to select the same on both tabs 
4. On the first tab, click on the "Archive" or "Restore" button. The value for "_Is archived_" should update itself 
5. On the second tab, the value for "_Is archived_" should have updated itself too
